lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -351,4 +352,78 @@ public Classification getClassificationByTask(Task task) throws ClassificationNo
         }
     }
 
+    @Override
+    public void deleteClassification(String classificationKey, String... domains) throws ClassificationInUseException, NotAuthorizedException, ClassificationNotFoundException {
+        if (domains.length == 0) {
+            throw new IllegalArgumentException("You have to specify a domain");
+        }
+        for (String domain : domains) {
+            if (domain.equals("")) {
+                // Alternative Solution: open "deleteRootClassification".
+                // But in my opinion there is a big difference between deleting a specification of a Classification and deleting the whole classification.
+                throw new IllegalArgumentException("You cannot delete the Root Classification this way. Please use 'deleteRootClassification' instead");
+            }
+        }
+
+        TaskServiceImpl taskService = (TaskServiceImpl) taskanaEngineImpl.getTaskService();
+        boolean existingTasksForClassificationInDomains = ((TaskServiceImpl) taskService).isThereATaskForThisClassificationKeyInTheseDomains(classificationKey, domains);
+        if (existingTasksForClassificationInDomains) {
+            throw new ClassificationInUseException("There are Tasks which belong to this classification. Please complete them and try again.");
+        }
+
+        this.internalDeleteClassification(classificationKey, domains);
```

Comment:
Here´s no try-catch but in the other method? Please make it constantly with one call-pattern.